microceph/cmd/microceph/disk_remove.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/canonical/microcluster/microcluster"
+	"github.com/spf13/cobra"
+
+	"github.com/canonical/microceph/microceph/api/types"
+	"github.com/canonical/microceph/microceph/client"
+)
+
+type cmdDiskRemove struct {
+	common *CmdControl
+	disk   *cmdDisk
+
+	flagBypassSafety     bool
+	flagConfirmDowngrade bool
+	flagTimeout          int64
+}
+
+func (c *cmdDiskRemove) Command() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "remove <osd-id> [--timeout=300] [--bypass-safety-checks=false] [--confirm-failure-domain-downgrade=false]",
+		Short: "Remove a Ceph disk (OSD) given as an integer ID.",
+		RunE:  c.Run,
+	}
+
+	cmd.PersistentFlags().Int64Var(&c.flagTimeout, "timeout", 300, "Timeout to wait for safe removal (seconds), default=300")
+	cmd.PersistentFlags().BoolVar(&c.flagBypassSafety, "bypass-safety-checks", false, "Bypass safety checks")
+	cmd.PersistentFlags().BoolVar(&c.flagConfirmDowngrade, "confirm-failure-domain-downgrade", false, "Confirm failure domain downgrade if required")
+
+	return cmd
+}
+
+func (c *cmdDiskRemove) Run(cmd *cobra.Command, args []string) error {
+	if len(args) != 1 {
+		return cmd.Help()
+	}
+
+	m, err := microcluster.App(context.Background(), microcluster.Args{StateDir: c.common.FlagStateDir, Verbose: c.common.FlagLogVerbose, Debug: c.common.FlagLogDebug})
+	if err != nil {
+		return err
+	}
+
+	cli, err := m.LocalClient()
+	if err != nil {
+		return err
+	}
+
+	osd, err := strconv.ParseInt(args[0], 10, 64)
```

Comment:
If a user is used to Ceph, they may be familiar with using the osd.$id reference rather than just the $id. Should both options be allowed? It's even referred to in the output below as osd.$id.